src/indra/fetch/cds.py

```
@@ -56,314 +63,465 @@ def last_date_of_cds_data(suppress_output=True):
+                logger.error("Access to CDS API is not authorized. Check your credentials.")
+                raise
+            elif "latest date available" in error_msg.lower() or error_msg.startswith("400"):
+                import re
+                match = re.search(r"(\d{4}-\d{2}-\d{2} \d{2}:\d{2})", error_msg)
+                if match:
+                    latest_timestamp = datetime.strptime(match.group(1), "%Y-%m-%d %H:%M")
+                    logger.info("Latest timestamp on CDS: %s", latest_timestamp.strftime("%Y-%m-%d %H:%M"))
+                    return latest_timestamp, std_op
+                else:
+                    logger.error("Failed to parse latest date from CDS error: %s", error_msg)
+                    raise ValueError("Could not find YYYY-MM-DD HH:MM in CDS API error response")
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)